### DISCUSSION TITLE
Future workflow printing experience

### DISCUSSION THREAD

**Discussion by hexylena** [galaxyproject/galaxy#17950] [Ideas]
I'm so excited for the new URL parameters on the workflow page (and the new workflow preview page!) `&embed=true&buttons=false&about=false&heading=false&minimap=false&zoom_controls=false`

These are going to make nicer views so much easier! After [some discussion](https://github.com/galaxyproject/training-material/pull/4553#issuecomment-1838449526) on the proposed tutorial using those features, it would be really nice if there was a more user-friendly way to access those. 

Namely a "print view" button (link). In the short term, just adding those parameters to the current URL would be really useful and a bit more discoverable with low cost (and people can ctrl-p or screenshot or whatever they like to do.)

In the long term @ElectronicBlueberry is right, a "print workflow" button that does server side rendering into a PNG would be an even better experience.

**Comment by hexylena** [comment]
@dannon what's the rationale/metric for turning these issues into discussions? (just to make sure I file them correctly) are all future requests going forward discussions? What discussion should happen here instead, design, implementation ideas for it?

**Comment by dannon** [comment]
It's admittedly kind of fuzzy, and this is also a bit of an experiment to see how people *want* to use discussions, but that's okay since there's zero cost to flipping them back and forth from issue to discussion

The way I'm thikning about it is that, for the most part, issues are things that are specifically actionable and have enough information available for any developer to be able to jump in, work on, and close the issue.  If we wanted to break out just the 'short term' bit above as an issue, for example, that'd make sense to me as an assignable unit (maybe that's the benchmark -- issues are things that can be readily assigned and tackled?).  The long term plan of how to generate these images if it runs as a hidden tool, a celery task, or something else, etc., could use discussion.

Again, totally an experiment, and we can use this however folks want.

**Reply by hexylena** [reply]
I love the experiment! (As data for you) We did a similar one in the GTN but it had very mixed success. I think it worked *best* when we had authors submitting individual threads in [the 'feedback' issue](https://github.com/galaxyproject/training-material/discussions/1452), where they would naïvely start a new thread (ideal) and we could reply under those threads (ideal.) or where we were asking for [specific reports of experiences.](https://github.com/galaxyproject/training-material/discussions/4151) But we've had less success getting people to suggest their preferred implementation/preferred UX or things we can then concretely go on to implement.

> are specifically actionable and have enough information available for any developer to be able to jump in, work on, and close the issue

actionability seems like a good metric. I guess I could've filed this more as "planning" or so, perhaps a few of those would make good discussions as well. 

(As for the long term I have pretty concrete opinions, it should use the same mechanisms of the history archive generation. but yes discussion on that, if people were to open new threads with their preferred implementation strategy so people could discuss those alternatives, it could work!)

> Again, totally an experiment, and we can use this however folks want.

sounds great! <del>I just want threading to be used properly :laughing: </del>
